angrmanagement/ui/views/code_view.py

```diff
@@ -41,16 +42,26 @@ def decompile(self):
         if self._function is None:
             return
 
-        d = self.workspace.instance.project.analyses.Decompiler(
-            self._function,
-            cfg=self.workspace.instance.cfg,
-            options=self._options.option_and_values,
-            optimization_passes=self._options.selected_passes,
-            peephole_optimizations=self._options.selected_peephole_opts,
-            # kb=dec_kb
-            )
-        self.codegen = d.codegen
-        self.set_codegen(d.codegen)
+        # retrieve saved codegens
+        try:
+            self.codegen = self.saved_codegen[self._function]
+        except KeyError:
+            self.codegen = None
+
+        if self.codegen is None:
+            d = self.workspace.instance.project.analyses.Decompiler(
+                self._function,
+                cfg=self.workspace.instance.cfg,
+                options=self._options.option_and_values,
+                optimization_passes=self._options.selected_passes,
+                peephole_optimizations=self._options.selected_peephole_opts,
+                # kb=dec_kb
+                )
+            self.codegen = d.codegen
+            # save a codegen so we don't need to reconstruct it again
+            self.saved_codegen[self._function] = self.codegen
+
+        self.set_codegen(self.codegen)
```

Comment:
This feature will conflict with my on-going development of decompilation manager. Wait until it is merged to master.

Reply:
@ltfish then this whole PR will need to wait since variable renaming, as well as syncing, is broken without some sort of decompilation caching